ros2param/ros2param/api/__init__.py

```diff
@@ -135,103 +63,34 @@ def load_parameter_dict(*, node, node_name, parameter_dict):
-                raise RuntimeError('Invalid structure of parameter file for node {}'
-                                   'expected same format as provided by ros2 param dump'
-                                   .format(k))
-            param_dict.update(value['ros__parameters'])
-        load_parameter_dict(node=node, node_name=node_name, parameter_dict=param_dict)
-
-
 def call_describe_parameters(*, node, node_name, parameter_names=None):
-    # create client
-    client = node.create_client(
-        DescribeParameters, f'{node_name}/describe_parameters')
-
-    # call as soon as ready
-    ready = client.wait_for_service(timeout_sec=5.0)
-    if not ready:
-        raise RuntimeError('Wait for service timed out')
-
-    request = DescribeParameters.Request()
-    if parameter_names:
-        request.names = parameter_names
-    future = client.call_async(request)
+    client = AsyncParameterClient(node, node_name)
+    future = client.describe_parameters(parameter_names)
     rclpy.spin_until_future_complete(node, future)
-
-    # handle response
     response = future.result()
     return response
 
 
 def call_get_parameters(*, node, node_name, parameter_names):
-    # create client
-    client = node.create_client(GetParameters, f'{node_name}/get_parameters')
-
-    # call as soon as ready
-    ready = client.wait_for_service(timeout_sec=5.0)
-    if not ready:
-        raise RuntimeError('Wait for service timed out')
-
-    request = GetParameters.Request()
-    request.names = parameter_names
-    future = client.call_async(request)
+    client = AsyncParameterClient(node, node_name)
+    future = client.get_parameters(parameter_names)
     rclpy.spin_until_future_complete(node, future)
-
-    # handle response
     response = future.result()
     return response
 
 
 def call_set_parameters(*, node, node_name, parameters):
-    # create client
-    client = node.create_client(SetParameters, f'{node_name}/set_parameters')
-
-    # call as soon as ready
-    ready = client.wait_for_service(timeout_sec=5.0)
-    if not ready:
-        raise RuntimeError('Wait for service timed out')
-
-    request = SetParameters.Request()
-    request.parameters = parameters
-    future = client.call_async(request)
+    client = AsyncParameterClient(node, node_name)
+    future = client.set_parameters(parameters)
     rclpy.spin_until_future_complete(node, future)
-
-    # handle response
     response = future.result()
     return response
 
 
 def call_list_parameters(*, node, node_name, prefix=None):
-    # create client
-    client = node.create_client(ListParameters, f'{node_name}/list_parameters')
-
-    # call as soon as ready
-    ready = client.wait_for_service(timeout_sec=5.0)
-    if not ready:
-        raise RuntimeError('Wait for service timed out')
-
-    request = ListParameters.Request()
-    future = client.call_async(request)
+    client = AsyncParameterClient(node, node_name)
+    future = client.list_parameters()
```

Comment:
Before the refactor, we were waiting for the service to be ready. Maybe we should wait for the services to be ready before calling `list_parameters()`?

Same question applies to all other service calls above.